core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1441,57 +1441,115 @@ public boolean removePartition(long timestamp) {
         if (!PartitionBy.isPartitioned(partitionBy)) {
             return false;
         }
-
         final long minTimestamp = txWriter.getMinTimestamp();
         final long maxTimestamp = txWriter.getMaxTimestamp();
-
         timestamp = getPartitionLo(timestamp);
         if (timestamp < getPartitionLo(minTimestamp) || timestamp > maxTimestamp) {
             LOG.error().$("partition is empty, folder does not exist [path=").$(path).I$();
             return false;
         }
 
-        if (timestamp == getPartitionLo(maxTimestamp)) {
-            LOG.error()
-                    .$("cannot remove active partition [path=").$(path)
-                    .$(", maxTimestamp=").$ts(maxTimestamp)
-                    .I$();
-            return false;
-        }
-
-        if (!txWriter.attachedPartitionsContains(timestamp)) {
-            LOG.error().$("partition is already detached [path=").$(path).I$();
+        final int index = txWriter.getPartitionIndex(timestamp);
+        if (index < 0) {
+            LOG.error().$("partition is already removed [path=").$(path).I$();
             return false;
         }
 
-        // when we want to delete first partition we must find out
-        // minTimestamp from next partition if it exists or next partition and so on
+        // when we want to delete first partition we must find out minTimestamp from
+        // next partition if it exists, or next partition, and so on
         //
-        // when somebody removed data directories manually and then
-        // attempts to tidy up metadata with logical partition delete
-        // we have to uphold the effort and re-compute table size and its minTimestamp from
-        // what remains on disk
+        // when somebody removed data directories manually and then attempts to tidy
+        // up metadata with logical partition delete we have to uphold the effort and
+        // re-compute table size and its minTimestamp from what remains on disk
 
-        // find out if we are removing min partition
-        long nextMinTimestamp = minTimestamp;
-        if (timestamp == txWriter.getPartitionTimestamp(0)) {
-            nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
-        }
-        long partitionNameTxn = txWriter.getPartitionNameTxnByPartitionTimestamp(timestamp);
-        columnVersionWriter.removePartition(timestamp);
+        if (timestamp == getPartitionLo(maxTimestamp)) {
+            // removing the active partition
+            rollback();
```

Comment:
in other places we do `commit()`, please have a look

Reply:
done. I also do commit, unless I do truncate.